pages/a/1001-node-crate-cluster-with-microsoft-azure.html

```diff
@@ -0,0 +1,185 @@
+title: A 1001 Node Crate Cluster with Microsoft Azure
+author: Chris Ward
+description: We show how to scale your Crate cluster for billions of records and beyond as simple as possible by launching a 1001 node cluster on Microsoft Azure infrastructure.
+created: 2016-03-30
+status: publish
+post_type: post
+tags: azure, docker, cloud
+category: developernews
+
+[Crate](https://crate.io/) is striving to be the most scalable database in our industry. We aim to make scaling your data from millions to billions of records and beyond as simple as possible.
+
+We have clients who write and query gigabytes of data a minute, store terabytes a day and love the performance and stability that Crate offers. But the question always remained at the back of our minds - when taken to the extreme, what is Crate really capable of?
+
+We specifically wanted to:
+
+- Deploy a Crate cluster of 1001 nodes.
+- Examine the overhead added by cluster management.
+- See what speed the cluster could write data.
+- Assess the performance of different cluster configurations.
+
+Microsoft Azure generously lent us the use of 8000 cores as the infrastructure for this test and engineer time to guide us through problems and questions we might have.
+
+Microsoft wanted to show that their Azure platform was reliable and capable enough for a project that would scale rapidly and is so resource intensive.
+
+## The Plan
+
+While discussing the approach of setting up a 1001 node Crate cluster, we immediately thought about using [Docker Swarm](https://www.docker.com/products/docker-swarm) to launch Crate. However, we also wanted to see how the conventional deployment, using regular packages, would work.
+
+### Docker Swarm Setup
+
+Our approach was to start the Crate cluster in a containerized environment using [Docker](https://www.docker.com), [Docker Swarm](https://www.docker.com/products/docker-swarm) for scheduling Crate instances and [Consul](https://www.consul.io/) for node discovery.
```

Comment:
we were using consul as storage backend for swarm, not for discovery. we were using 3 master nodes and they were listed in DNS SRV records (@chaudum knows)